metaflow/plugins/argo/argo_workflow.py

```diff
@@ -0,0 +1,586 @@
+            return child_foreach_task(self.node)
+        return regular_task(self.node)
+
+    def template(self):
+        tmpl = {
+            'name': self.name,
+            'inputs': self._inputs(),
+            'outputs': self._outputs()
+        }
+        metadata = {k: v for k, v in self._attr.items()
+                    if k in ('labels', 'annotations') and v}
+        if metadata:
+            tmpl['metadata'] = metadata
+        if self._attr.get('nodeSelector'):
+            tmpl['nodeSelector'] = self._attr['nodeSelector']
+        tmpl['container'] = self._container()
+        return tmpl
+
+    def _inputs(self):
+        params = ['input-paths']
+        if self.is_nested_foreach():
+            params.append('split-index')
+        inputs = {
+            'parameters': [{'name': ip} for ip in params]
+        }
+        artifacts = self._attr.get('input_artifacts')
+        if artifacts:
+            inputs['artifacts'] = artifacts
+        return inputs
+
+    def _outputs(self):
+        params = [{
+            'name': 'task-id',
+            'value': '{{pod.name}}'
+        }]
+        if self.node.type == 'foreach':
+            params.append({
+                'name': 'num-splits',
+                'valueFrom': {'path': ArgoInternalStepDecorator.splits_file_path}
+            })
+        outputs = {'parameters': params}
+        artifacts = self._attr.get('artifacts', []) + self._attr.get('output_artifacts', [])
+        if artifacts:
+            outputs['artifacts'] = artifacts
+        return outputs
+
+    def _container(self):
+        image = self.default_image
+        if self._attr.get('image'):
+            image = self._attr['image']
+        env, env_from = self._prepare_environment()
+        container = {
+            'image': image,
+            'command': ['/bin/sh'],
+            'args': ['-c', ' && '.join(self.cmds)],
+            'env': env,
+        }
+        if env_from:
+            container['envFrom'] = env_from
+        res = self._resources()
+        if res:
+            container['resources'] = {
+                'requests': res,
+                'limits': res
+            }
+        return container
+
+    def _resources(self):
+        res = {}
+        cpu = self._attr.get('cpu')
+        if cpu:
+            res['cpu'] = int(cpu)
+        mem = self._attr.get('memory')
+        if mem:
+            # argo cluster treats memory as kb
+            res['memory'] = str(mem) + 'Mi'
+        gpu = self._attr.get('gpu')
+        if gpu:
+            res['nvidia.com/gpu'] = int(gpu)
+        return res
+
+    def _prepare_environment(self):
+        default = {
+            'METAFLOW_USER': get_username(),
+            'METAFLOW_DATASTORE_SYSROOT_S3': DATASTORE_SYSROOT_S3,
+        }
+        if METADATA_SERVICE_URL:
+            default['METAFLOW_SERVICE_URL'] = METADATA_SERVICE_URL
+        if DEFAULT_METADATA:
+            default['METAFLOW_DEFAULT_METADATA'] = DEFAULT_METADATA
```

Comment:
`METAFLOW_SERVICE_HEADERS` is another one we'd need to pass through

Reply:
sure

Reply:
done